src/main/java/net/neoforged/neoforge/common/crafting/outgredient/TagDefaultsManager.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.common.crafting.outgredient;
```

Comment:
IMO, a tag "defaults" manager belongs in its own package as it has uses beyond recipe results. You mentioned loot table outputs as one of them; which I would recommend not doing via your new "outgredient" but just adding a new custom loot table entry type that queries this manager.

Reply:
I agree. Another use case could be with fluids, like in the Tinkers' Construct smeltery.